### ISSUE TITLE
Confusing error message pertaining to deprecation of "value" attribute of cloudflare_record

### ISSUE THREAD

**Issue by mccartney** [cloudflare/terraform-provider-cloudflare#3590]
### Confirmation

- [X] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [X] I have searched the issue tracker and my issue isn't already found.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
terraform -v
Terraform v1.5.7
on darwin_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.39.0
+ provider registry.terraform.io/hashicorp/aws v5.62.0
```

### Affected resource(s)

cloudflare_record

### Terraform configuration files

```hcl
resource "cloudflare_record" "CNAME_my" {
  zone_id = cloudflare_zone.my.id
  name    = "o-c.kr"
  type    = "CNAME"
  ttl     = 7200
  value   = "REDACTED.amazonaws.com."
}
```

### Link to debug output

N/A

### Expected output

- If `value` is deprecated, I assume it's still legal to use it. So I assume the code should work, i.e. `terraform apply` should succeed.
- Also the error message is misleading. It claims I have provided value for `content`, which I didn't.

### Actual output

│ Error: either 'content' (present: true) or 'data' (present: false) must be provided

### Steps to reproduce

terraform apply

**Comment by mccartney** [comment]
> This issue has been marked with triage/needs-information and is unlikely to receive maintainer attention until the log file is provided making this a complete bug report.

Fair enough.
I did my best to provide you valuable input. I don't think `TF_LOG=DEBUG` is needed in this specific case.

**Comment by mccartney** [comment]
Related to #3509 
(which is BTW funny as it's an anagram of this issue id)

**Comment by mccartney** [comment]
Works fine after downgrading to `4.38.0`.

**Comment by kofshikz** [comment]
It appears that just changing `value` to `content` also helps

**Comment by mccartney** [comment]
> Works fine after downgrading to 4.38.0.

Watch out when/if downgrading. If you had some runs with `4.39.0` it might have changed `schema_version` from `2` to `3` in your Terraform state file for existing records. In such a case, Terraform will refuse to work with your resources after downgrading with messages like:
```
│ The current state of cloudflare_record.REDACTED was created by a newer provider version than is currently selected. Upgrade the cloudflare provider to work with this state.
```
I am reverting the state file from S3 versioned objects because of this.

So, I haven't tried it, but it looks like the suggestion from @kofshikz might actually be a safer choice than downgrading.

**Comment by erezhazan1** [comment]
> It appears that just changing `value` to `content` also helps

yes, this helped me as well

**Comment by jacobbednarz** [comment]
the error message has already been updated in #3580. if your haven't already, i recommend checking out the CHANGELOG.md and release notes which call out how to do this migration.

**Comment by Engerim** [comment]
But that is not a deprecation, you are deprecating `value` and in the same pr it is impossible to use `value` anymore so it is a major change and breaking, as with 4.39.0 value can not be used anymore, so that should be a major release instead of a minor

**Comment by cailen** [comment]
I agree with the sentiment here. This is not a deprecation. It is a replacement. The maintainers of this provider have for too long been playing fast and loose with the term "deprecation" and I feel sorry for anyone that is going to run into this one because it is a doozy.

**Comment by cailen** [comment]
> the error message has already been updated in #3580. if your haven't already, i recommend checking out the CHANGELOG.md and release notes which call out how to do this migration.

@jacobbednarz I don't see how that addresses the issue here: you are saying it's ok to have `content` or `data`, but still are not addressing that `value` can no longer be used, which isn't a deprecation, but a removal IMO. This is what it generates currently un 4.39.0:

```
│ Warning: Argument is deprecated
│ 
│   with module.cloudflare.cloudflare_record.optional["www"],
│   on .terraform/modules/cloudflare/main.tf line 66, in resource "cloudflare_record" "optional":
│   66:   value   = each.value.value
│ 
│ `value` is deprecated in favour of `content` and will be removed in the next major release.
╵
╷
│ Error: either 'content' (present: true) or 'data' (present: false) must be provided
│ 
│   with module.cloudflare.cloudflare_record.optional["www"],
│   on .terraform/modules/cloudflare/main.tf line 59, in resource "cloudflare_record" "optional":
│   59: resource "cloudflare_record" "optional" {
│ 
```

This says `value` is deprecated in favor of content and will be removed in the next major release, but you cannot at the present use `value` at all.

**Comment by cailen** [comment]
@jacobbednarz I also ask that this issue be reopened as I don't think it has been properly addressed.

**Comment by ido-gloat** [comment]
This is a major bug!
I can't you my tf code and can't downgrade  the version because my other moduled in higher version

**Comment by georgekaz** [comment]
Wow, I am shocked that a change like this got through QA. You can't just change the spec of cloudflare_record that's so fundamental via a patch release. Rethink or revert.

**Comment by mgirouard** [comment]
Hey gang, I've cut a PR which I believe should give us a path forward. Please see #3674.

**Comment by mgirouard** [comment]
Howdy, we've just merged #3674 to `master` and will prepare a release in the next day or two. If you have the ability to test on an unreleased HEAD, please do so and let us know if you ran into any trouble.

To quickly summarize the patch:
- `value` will remain deprecated but still usable (addresses this issue); you will continue to see the deprecation warnings so please update to use `content` before moving to the next major release.
- `value` and `content` are mutually exclusive. Setting both will return an error.
- There will be no state changes when changing to `content`. This this is a safe operation and should be a no-op when you run `terraform plan` again.

**Comment by mccartney** [comment]
Interestingly (at least in `4.39.0`) when you use `value=` instead of `content=` the problem is only caught during `terraform apply`, not `terraform plan`:
```
06:40:13  │ Error: either 'content' (present: true) or 'data' (present: false) must be provided
```
